tests/test_threadpool_limits.py

```diff
@@ -0,0 +1,193 @@
+import ctypes
+import pytest
+
+
+from loky import cpu_count
+from loky.backend import thread_pool_limits
+from loky.backend._thread_pool_limiters import _CLibsWrapper
+from loky.backend._thread_pool_limiters import NOT_ACCESSIBLE
+from loky.backend._thread_pool_limiters import get_thread_limits
+from loky.backend._thread_pool_limiters import _set_thread_limits
+from loky.backend._thread_pool_limiters import openmp_num_threads
+
+from .utils import with_parallel_sum, libopenblas_paths
+
+
+def should_skip_module(module):
+    return module['name'] == "openblas" and module['version'] == NOT_ACCESSIBLE
+
+
+def test_openmp_num_threads():
+    with pytest.raises(ValueError, match=r"n_threads=0 is invalid"):
+        openmp_num_threads(0)
+
+    assert openmp_num_threads(1) == 1
+    assert openmp_num_threads(2) == 2
+    assert openmp_num_threads(3) == 3
+
+    # Assert negative value give the right number
+    assert openmp_num_threads(-1) == cpu_count()
+    assert openmp_num_threads(-2) == max(1, cpu_count() - 1)
+    assert openmp_num_threads(-3) == max(1, cpu_count() - 2)
+
+
+@pytest.mark.parametrize("library", _CLibsWrapper.SUPPORTED_CLIBS)
+def test_thread_pool_limits(openblas_test_noskip, mkl_win32_test_noskip,
+                            library):
+    old_limits = get_thread_limits()
+    old_limits = {clib['name']: clib['n_thread'] for clib in old_limits}
+
+    if library not in old_limits:
+        if library == "openblas" and openblas_test_noskip:
+            raise RuntimeError("Could not load the OpenBLAS library")
+        elif library == "mkl_win32" and mkl_win32_test_noskip:
+            import numpy as np
+            np.dot(np.ones(1000), np.ones(1000))
+            old_limits = get_thread_limits()
+            if old_limits[library] is None:
+                raise RuntimeError("Could not load the MKL library")
+        else:
+            pytest.skip("Need {} support".format(library))
+
+    new_limits = _set_thread_limits(limits={library: 1})
+    new_limits = {clib['name']: clib['n_thread'] for clib in new_limits}
+    assert new_limits[library] == 1
```

Comment:
Please add a case that checks for `_set_thread_limits(limits={library: None})` and checks that everything is reset to something consistent with `old_limits`.

Reply:
Don't we said that `_set_thread_limits` should not accept `None` anymore?

Reply:
The reset option seems pretty complicated to me as I am not sure where to store the original values. To me, this functionality should be used with the context manager to avoid messing the setup anywhere else.

Reply:
Yes you are right. I wrote this comment prior to changing my mind on `_set_thread_limits(limits={library: None})`. Let's use the context manager for that use case.